### PR TITLE
Add a grammar test for TSQL

### PR DESCRIPTION
**Pull request by lppedd**
This PR adds a test for the TSQL grammar, probably the largest grammar you can find in the ANTLR Grammars repository.

The generated file is so big it causes a `o.j.o.objectweb.asm.MethodTooLargeException` from the Kotlin compiler when targeting the JVM, so the testing is enabled only for the JS platform for now.

I will report `MethodTooLargeException` in YouTrack soon.

### CODE REVIEW

**Comment by lppedd** [comment]
As expected it is failing on the JVM. Let's keep this PR parked.

**Comment by lppedd** [comment]
@KvanTTT FYI

**Review by KvanTTT — COMMENTED** [review]
Thanks, but I'm afraid it's JVM limitation but not Kotlin itself.

Also, thanks for the PR, but I think it's better to use generated grammar and data for this purpose. Currently we have such kind of tests in [CustomDescriptors.java](https://github.com/antlr/antlr4/blob/dev/runtime-testsuite/test/org/antlr/v4/test/runtime/CustomDescriptors.java) (official ANTLR repository).

**Comment by lppedd** [comment]
@ftomassetti to recap the commits here:

- first commit adds a JS and a Native grammar test for the TSQL grammar.  
  The test cannot be common because on the JVM compilation fails with `MethodTooLargeException`
- second commit filter out the `TSql*` generated files from the JVM compilation, since we don't need them.  
  This means the other phases now completes correctly.
- third commit fixes a strange compiler error that happens only with the TSQL grammar.  
  It probably depends on the size of the compiled code.

I think we can merge as testing with grammar this large will save us from many other issues in the future.

File: antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/RuleContext.kt [hunk 1]
```diff
@@ -119,7 +119,7 @@ public open class RuleContext : RuleNode {
       return builder.toString()
     }
 
-  public open var ruleIndex: Int = -1
+  public open val ruleIndex: Int = -1
```
Comment (ftomassetti) [inline antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/RuleContext.kt:122]: Should we introduce a constant called `INVALID_RULE_INDEX` or `UNASSIGNED_RULE_INDEX` set to 1 and refer to that constant here and in the constructor of `InterpreterRuleContext`?

File: antlr-kotlin-tests/src/jsTest/kotlin/com/strumenta/antlrkotlin/test/grammars/TSqlGrammarJsTest.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import com.strumenta.antlrkotlin.test.generated.TSqlParser
+import org.antlr.v4.kotlinruntime.CharStream
+import org.antlr.v4.kotlinruntime.ParserRuleContext
+import org.antlr.v4.kotlinruntime.TokenStream
+
+// Note(Edoardo): move the test back to commonTest once
+//  the MethodTooLargeException has been solved on the JVM
+//  by the Kotlin team.
+//  FQN: org.jetbrains.org.objectweb.asm.MethodTooLargeException
+@Suppress("unused")
```
Comment (ftomassetti) [inline antlr-kotlin-tests/src/jsTest/kotlin/com/strumenta/antlrkotlin/test/grammars/TSqlGrammarJsTest.kt:18]: Should we add a link to a YouTrack issue here?

**Comment by ftomassetti** [comment]
Thank you @lppedd and @KvanTTT 

I added minor comments

File: antlr-kotlin-tests/src/jsTest/kotlin/com/strumenta/antlrkotlin/test/grammars/TSqlGrammarJsTest.kt (hunk 2 above)
Comment (lppedd) [reply antlr-kotlin-tests/src/jsTest/kotlin/com/strumenta/antlrkotlin/test/grammars/TSqlGrammarJsTest.kt:18]: I still haven't created it, but I think we can add it later on anyway.

File: antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/RuleContext.kt (hunk 1 above)
Comment (ftomassetti) [reply antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/RuleContext.kt:122]: Make sense. I would then consider adding a comment instead.
Perhaps we could add an issue about that in the antlr4 repo, but that is definitely out of the scope of this PR

File: antlr-kotlin-tests/src/jsTest/kotlin/com/strumenta/antlrkotlin/test/grammars/TSqlGrammarJsTest.kt (hunk 2 above)
Comment (ftomassetti) [reply antlr-kotlin-tests/src/jsTest/kotlin/com/strumenta/antlrkotlin/test/grammars/TSqlGrammarJsTest.kt:18]: Makes sense

File: antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/RuleContext.kt (hunk 1 above)
Comment (lppedd) [reply antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/RuleContext.kt:127]: @ftomassetti addressed!

Comment (ftomassetti) [reply antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/RuleContext.kt:127]: Thank you!

**Pull request merged**